### PR TITLE
Introduce new top level test directory modules

### PR DESCRIPTION
**Pull request by clacroix12**
Jira: https://issues.redhat.com/browse/OCSQE-839

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

**Review by petr-balogh — DISMISSED** [review]

**Review by vavuthu — DISMISSED** [review]

File: tests/functional/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Functional Testing
+Tests that are focused on one particular component or functionality of OCS will belong in the Functional module. Some examples could be object testing, monitoring testing, pv testing, deployment or upgrade tests, etc.
```
Comment (ebenahar) [inline tests/functional/README.md:2]: same here

**Comment by clacroix12** [comment]
@ebenahar I fixed the references to OCS

@mbukatov I have expanded a bit on the functional vs. cross functional bit, given some specific examples, explained why they fit into those categoriers, and also included a section to help people decide where their test cases should go. 

If this is still missing some explanation or depth please let me know and I can elaborate a bit more.

**Comment by clacroix12** [comment]
@ebenahar @mbukatov @red-hat-storage/ocs-qe-ecosystem Could still use a review of this so we can get the new directories merged.

**Review by petr-balogh — APPROVED** [review]
LGTM

**Comment by petr-balogh** [comment]
@clacroix12 only one thing I am afraid of that if people will start moving the test cases to the new structure. And as we didn't yet changed the squad marker which is defined currently here:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/constants.py#L1594-L1605

Based on path, this will need to be addressed as well to make this happen on new test path base or we need to have this decorated by user in the module itself marking the specific test case. Or what did we agree on to do here?

**Comment by clacroix12** [comment]
Marking this as WIP as we need to implement our new mechanism of test case ownership prior to moving test cases.

**Comment by ebenahar** [comment]
@clacroix12 Would like to see how we can resume the work here

**Review by shivamdurgbuns — APPROVED** [review]

**Comment by clacroix12** [comment]
We have implemented squad ownership via decorators (#8483) and the team has been notified of this requirement moving forward. We can now move forward with adding the new top level directories for test cases.

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+An example of this kind of testing is our [mcg](https://github.com/red-hat-storage/ocs-ci/tree/master/tests/manage/mcg) testing. This module is aimed at testing one ODF component (MCG) and all of the tests in this module are aimed at testing one particular feature of that component (bucket_creation, multi_region, object_versioning, etc.).
+
+### Cross Functional
+Tests that target multiple components or perform actions that span different areas of the product will belong in the Cross Functional module.
+
+An example of this kind of testing is our [longevity](https://github.com/red-hat-storage/ocs-ci/tree/master/tests/e2e/longevity) testing. These test cases are not targeting one specific feature of the product. Rather, they are creating scenarios that represent real life user behavior which spans across several different product features/components. Each test case under this module performs very different actions from the others with regards to which components are involved. See [test_stage1.py](https://github.com/red-hat-storage/ocs-ci/blob/master/tests/e2e/longevity/test_stage1.py) for a specific example.
```
Comment (ebenahar) [inline tests/README.md:16]: would the link path of https://github.com/red-hat-storage/ocs-ci/tree/master/tests/e2e/longevity change once we do the restructure?

**Pull request merged**